docs/API/v3/search-query-service-resource.md

```diff
@@ -0,0 +1,151 @@
+keywords: NuGet V3 search API, NuGet V3 discover packages, API to query NuGet packages, API to browse NuGet packages
+ms.reviewer:
+- karann
+- unniravindranathan
+
+---
+
+# Search
+
+It is possible to search for packages available on a package source using the V3 API. The resource used for searching
+is the `SearchQueryService` resource found in the [service index](service-index.md).
+
+## Versioning
+
+There are three resource `@type` values for searching. All three values represent the exact same API surface area. Note
+that non-breaking changes (defined as added JSON properties or query parameters) are periodically added and documented
+here.
+
+The `@type` values are:
+
+ - `SearchQueryService`
+ - `SearchQueryService/3.0.0-beta`
+ - `SearchQueryService/3.0.0-rc`
+
+### API changes
+
+ - August 2017: Added the `verified` property to the search result.
+ - March 2017: Added a `semVerLevel` query parameter which defaults to filtering out SemVer 2.0.0 packages.
+
+## Base URL
+
+The base URL for the following API is the value of the `@id` property associated with one of the aforementioned
+resource `@type` values. In the following document, this placeholder URL will be used:
+
+```
+https://example/query
+```
+
+## HTTP methods
+
+All URLs found in the search resource only support the HTTP method `GET`.
+
+## Search for packages
+
+The search API allows a client to query for a page of packages matching a specified search query. The interpretation
+of the search query (e.g. the tokenization of the search terms) is determined by the server implementation but the
+general expectation is that the search query is used for matching package IDs, titles, descriptions, and tags. Other
+package metadata fields may also be considered.
+
+An unlisted package should never appear in search results.
+
+```
+GET https://example/query?q={QUERY}&skip={SKIP}&take={TAKE}&prerelease={PRERELEASE}&semVerLevel={SEMVERLEVEL}
+```
+
+### Request parameters
+
+Name        | In     | Type    | Required | Notes
+----------- | ------ | ------- | -------- | -----
+q           | URL    | string  | no       | The search terms to used to filter packages
+skip        | URL    | integer | no       | The number of results to skip, for pagination
+take        | URL    | integer | no       | The number of results to return, for pagination
+prerelease  | URL    | boolean | no       | `true` or `false` determining whether to include [pre-release packages](../../create-packages/prerelease-packages.md)
+semVerLevel | URL    | string  | no       | A SemVer 2.0.0 version string 
+
+The search query `q` is parsed in a manner that is defined by the server implementation. nuget.org supports basic
+filtering on a [variety of fields](../../consume-packages/finding-and-choosing-packages.md#search-syntax). If no
+`q` is provided, all packages should be returned. This enables the "Browse" tab in the NuGet Visual Studio
```

Comment:
`all packages should be returned` seems like a dangerous statement :) perhaps change to `all packages, within boundaries imposed by skip and take - or their server default values when absent, should be returned`.

Reply:
Yeah, good point.

Reply:
Fixed.